spring-cloud-task-core/src/test/java/org/springframework/cloud/task/configuration/RepositoryTransactionManagerConfigurationTests.java

```diff
@@ -0,0 +1,237 @@
+		testConfiguration(MultipleTransactionManagerConfiguration.class);
+	}
+
+	private void testConfiguration(Class configurationClass) {
+		ApplicationContextRunner applicationContextRunner = new ApplicationContextRunner()
+			.withConfiguration(
+				AutoConfigurations.of(PropertyPlaceholderAutoConfiguration.class,
+					SimpleTaskAutoConfiguration.class,
+					configurationClass))
+			.withPropertyValues("application.name=transactionManagerTask");
+
+		applicationContextRunner.run((context) -> {
+			DataSource dataSource = context.getBean("dataSource", DataSource.class);
+
+			int taskExecutionCount = JdbcTestUtils
+				.countRowsInTable(new JdbcTemplate(dataSource), "TASK_EXECUTION");
+
+			// Verify that the create call was rolled back
+			assertThat(taskExecutionCount).isEqualTo(0);
+
+			// Execute a new create call so that things close cleanly
+			TaskRepository taskRepository = context.getBean("taskRepository",
+				TaskRepository.class);
+
+			TaskExecution taskExecution = taskRepository
+				.createTaskExecution("transactionManagerTask");
+			taskExecution = taskRepository.startTaskExecution(2, "transactionManagerTask",
+				new Date(), new ArrayList<>(0), null);
+
+			TaskLifecycleListener listener = context.getBean(TaskLifecycleListener.class);
+
+			ReflectionTestUtils.setField(listener, "taskExecution", taskExecution);
+		});
+	}
+
+	@EnableTask
+	@Configuration
+	public static class ZeroTransactionManagerConfiguration {
+
+		@Bean
+		public TaskConfigurer taskConfigurer(DataSource dataSource) {
+			return new DefaultTaskConfigurer(dataSource);
+		}
+
+		@Bean
+		public DataSource dataSource() {
+			DataSourceBuilder dsb = DataSourceBuilder.create()
+				.url("jdbc:h2:mem:testdb;DB_CLOSE_ON_EXIT=FALSE");
+			dsb.driverClassName("org.h2.Driver");
+			return dsb.build();
+		}
+	}
+
+	@EnableTask
+	@Configuration
+	public static class SingleTransactionManagerConfiguration {
+
+		@Bean
+		public TaskConfigurer taskConfigurer(DataSource dataSource,
+				PlatformTransactionManager transactionManager) {
+			return new DefaultTaskConfigurer(dataSource) {
+				@Override
+				public PlatformTransactionManager getTransactionManager() {
+					return transactionManager;
+				}
+			};
+		}
+
+		@Bean
+		public DataSource dataSource() {
+			DataSourceBuilder dsb = DataSourceBuilder.create()
+					.url("jdbc:h2:mem:testdb;DB_CLOSE_ON_EXIT=FALSE");
+			dsb.driverClassName("org.h2.Driver");
+			return dsb.build();
+		}
+
+		@Bean
+		public DataSourceTransactionManager transactionManager(DataSource ds) {
+			return new DataSourceTransactionManager(ds) {
+
+				private int count = 0;
+
+				@Override
+				protected void doCommit(DefaultTransactionStatus status) {
+
+					if (count == 0) {
+						// Rollback the finish of the task
+						super.doRollback(status);
+					}
+					else {
+						// Commit the start of the task
+						super.doCommit(status);
+					}
+
+					count++;
+				}
+			};
```

Comment:
We can probably move this to a static class because it is repeated in lines 201-219.   I can do this on merge.